### PR TITLE
Add `swagger-jsdoc` dependency and basic annotations

### PR DESCRIPTION
**Pull request by Green-Ranger11**
This adds `swagger-jsdoc` to auto-generate spec file via running `npm run openapi`. This will also add a swagger-ui.

Things left to do:

- [x] Add descriptions for each endpoint
- [x] Define schema definitions
- [x] Configure tags for each controller
- [x] Add command to generate spec file
- [x] Configure the auth middleware
- [x] Add swagger UI

### CODE REVIEW

File: openapi.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as fs from 'fs'
+
+import swaggerJSDoc from 'swagger-jsdoc';
+import swaggerJSDocOptions from './swaggerJSDocOptions.js';
+
+const openapiSpecification = swaggerJSDoc(swaggerJSDocOptions);
+
+fs.writeFileSync("./api-doc.json", JSON.stringify(openapiSpecification));
+
+console.log("Openapi spec generated in api-doc.json");
```
Comment (nfebe) [inline openapi.js]: Add a newline

File: routes/web.js [hunk 2]
```diff
@@ -10,36 +10,254 @@ import { body } from 'express-validator'
...
+ *       username:
+ *         type: string
+ *       password:
+ *         type: string
+ *       path:
+ *         type: string
+ */
+
+/**
+ * @openapi
```
Comment (nfebe) [inline routes/web.js]: Can you put this definition along side the other ones in swaggerJSDocOptions?

**Review by nfebe — CHANGES_REQUESTED** [review]
Thanks @Green-Ranger11 the looks pretty good for the start! Just some two minor requests and then I would like you to add a section in the `README` called #Docs with a subsection ##OpenAPI and there you write some notes (two to three lines) about how to:

- generate the json file
- what endpoint to visit to see the docs

cheers!

File: openapi.js (hunk 1 above)
Comment (Green-Ranger11) [reply openapi.js]: Will do!

File: routes/web.js (hunk 2 above)
Comment (Green-Ranger11) [reply routes/web.js]: Sure thing!

**Comment by Green-Ranger11** [comment]
Will definitely do that shortly thanks!

**Pull request merged**